src/lib/profiles/data-management/Current/ResourceIdentifier.cpp

```diff
@@ -237,6 +237,9 @@ WEAVE_ERROR ResourceIdentifier::FromString(const char * inBuffer, size_t bufferL
 
         if ((strncmp(resourceName, inBuffer, strlen(resourceName)) == 0) && inBuffer[strlen(resourceName)] == '_')
```

Comment:
Is the compiler smart enough in all cases to optimize all of the 'strlen(resourceName)' calls into one. At minimum, it might certainly make the code clearer if that was all calculated once and saved in a `const size_t`.

Reply:
Done.